File: 프로그래머스/3/42884. 단속카메라/단속카메라.java
import java.util.*;

class Solution {
    public int solution(int[][] routes) {
        Arrays.sort(routes, (a, b) -> Integer.compare(a[1], b[1]));
        
        int res = 1;
        int lastCheck = routes[0][1];
        
        for (int i = 1; i < routes.length; i++) {
            int start = routes[i][0];
            
            if (start > lastCheck) {
                lastCheck = routes[i][1];
                res++;
            }
        }
        return res;
    }
}

File: 프로그래머스/3/42884. 단속카메라/단속카메라.cpp
#include <bits/stdc++.h>
using namespace std;

bool compare(vector<int>& a, vector<int>& b) {
    return a[1] < b[1];
}

int solution(vector<vector<int>> routes) {
    int res = 0;
    int N = routes.size();
    
    sort(routes.begin(), routes.end(), compare);
    int start = routes[0][0];
    int lastCheck = routes[0][1];
    res++;
    
    for (int idx = 1; idx < N; idx++) {
        start = routes[idx][0];
        
        if (start <= lastCheck)
            continue;
        
        lastCheck = routes[idx][1];
        res++;
    }
    
    return res;
}